docs/Training-ML-Agents.md

```diff
@@ -1,196 +1,116 @@
 ```
 
-During a training session, the training program prints out and saves updates at
-regular intervals (specified by the `summary_freq` option). The saved statistics
-are grouped by the `run-id` value so you should assign a unique id to each
-training run if you plan to view the statistics. You can view these statistics
-using TensorBoard during or after training by running the following command:
-
-```sh
-tensorboard --logdir=summaries --port 6006
-```
+where
 
-And then opening the URL: [localhost:6006](http://localhost:6006).
+* `<trainer-config-file>` is the file path of the trainer configuration yaml. This contains all the
+  hyperparameter values. We offer a detailed guide on the structure of this file and the meaning
+  of the hyperameters (and advice on how to set them) in the dedicated
+  [Training Config File](#training-config-file) section below.
+* `<env_name>`__(Optional)__ is the name (including path) of your [Unity
+  executable](Learning-Environment-Executable.md) containing the agents to be trained.
+  If `<env_name>` is not passed, the training will happen in the Editor.
+  Press the :arrow_forward: button in Unity when the message _"Start training by
+  pressing the Play button in the Unity Editor"_ is displayed on the screen.
+* `<run-identifier>` is a unique name you can use to identify the results of your training runs.
+
+See the [Getting Started Guide](Getting-Started.md#training-a-new-model-with-reinforcement-learning)
+for a sample execution of the `mlagents-learn` command.
+
+#### Observing Training
+
+Regardless of which training methods, configurations or hyperparameters you provide,
+the training process will always generate three artifacts:
+1. Summaries (under the `summaries/` folder): these are training metrics that are updated
+throughout the training process. They are helpful to monitor your training performance
+and may help inform how to update your hyperparameter values.
+See [Using TensorBoard](Using-Tensorboard.md) for more details on how to visualize
+the training metrics.
+1. Models (under the `models/` folder): these contain the model checkpoints that are updated
+throughout training and the final model file (`.nn`). This final model file is generated once
+either when training completes or is interrupted.
+1. Timers file (also under the `summaries/` folder): this contains aggregated metrics on your
+training process, including time spent on specific code blocks.
+See [Profiling in Python](Profiling-Python.md) for more information on the timers generated.
+
+These artifacts (except the `.nn` file) are updated throughout the training process and finalized
+when training completes or is interrupted.
+
+#### Debugging
 
-**Note:** The default port TensorBoard uses is 6006. If there is an existing session
-running on port 6006 a new session can be launched on an open port using the --port
-option.
+If you enable the `--debug` flag in the command line, the trainer metrics are logged to a CSV file
```

Comment:
Unfortunately none of these things are correct anymore, I would just remove the whole Debugging section. 

The CSV is still created under the `summaries/` directory, but is slated to be deprecated and replaced with a JSON file, so it's probably OK if we don't mention it.

Reply:
So should we remove now and bring back once we add the JSON file?

Reply:
And also, remove the --debug option from the CLI?

Reply:
+1 for remove now and bring back once we add the JSON file. 

The `--debug` CLI option is still there, but it doesn't really help the user in debugging training (it's mostly for us), nor does it log the CSV. So it probably doesn't go in this doc. Is it OK if we just leave it to the `--help` menu to describe?